workers/unity/Packages/com.improbable.gdk.core/Worker/ViewDiff.cs

```diff
@@ -0,0 +1,307 @@
+                        componentIdToComponentStorage.Add(instance.GetComponentId(), instance);
+                    }
+
+                    if (typeof(ICommandDiffStorage).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (ICommandDiffStorage) Activator.CreateInstance(type);
+
+                        commandStorageList.Add(instance);
+                        if (!componentIdToCommandIdToStorage.TryGetValue(instance.GetComponentId(),
+                            out var commandIdToStorage))
+                        {
+                            commandIdToStorage = new Dictionary<uint, ICommandDiffStorage>();
+                            componentIdToCommandIdToStorage.Add(instance.GetComponentId(), commandIdToStorage);
+                        }
+
+                        commandIdToStorage.Add(instance.GetCommandId(), instance);
+                    }
+                }
+            }
+        }
+
+        public void Clean()
+        {
+            foreach (var storage in componentStorageList)
+            {
+                storage.Clear();
+            }
+
+            foreach (var storage in commandStorageList)
+            {
+                storage.Clear();
+            }
+
+            worldCommandStorage.Clear();
+            entitiesAdded.Clear();
+            entitiesRemoved.Clear();
+            logs.Clear();
+            metrics.Clear();
+            flags.Clear();
+        }
+
+        public void AddEntity(long entityId)
+        {
+            if (!entitiesRemoved.Remove(new EntityId(entityId)))
+            {
+                entitiesAdded.Add(new EntityId(entityId));
+            }
+        }
+
+        public void RemoveEntity(long entityId)
+        {
+            if (!entitiesAdded.Remove(new EntityId(entityId)))
+            {
+                entitiesRemoved.Add(new EntityId(entityId));
+            }
+        }
+
+        public void AddComponent<T>(T component, long entityId, uint componentId) where T : ISpatialComponentUpdate
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component ID");
+            }
+
+            ((IDiffComponentAddedStorage<T>) storage).AddEntityComponent(entityId, component);
+        }
+
+        public void RemoveComponent(long entityId, uint componentId)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component ID");
+            }
+
+            storage.RemoveEntityComponent(entityId);
+        }
+
+        public void SetAuthority(long entityId, uint componentId, Authority authority)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var authorityStorage))
+            {
+                throw new ArgumentException("Unknown component");
```

Comment:
Shouldn't this exception be the same as the others?

Reply:
The exceptions do miss some information overall, ArgumentException should have the ability to pass in which argument was wrong, and what its value was

Reply:
Added real error messages. I'm generally against putting interpolation in an exception but will do until we decide on something better.